### PR TITLE
[FEATURE] Add an option to set the default project file format (qgs/qgz)

### PR DESCRIPTION
**Pull request by 3nids**
This adds a new global option to define the preferred project file format

image

sponsored by GEM foundation

### CODE REVIEW

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -6660,12 +6660,23 @@ bool QgisApp::fileSave()
 
     const QString qgsExt = tr( "QGIS files" ) + " (*.qgs)";
     const QString zipExt = tr( "QGZ files" ) + " (*.qgz)";
+
+    QString exts;
+    QgsProject::FileFormat defaultProjectFileFormat = settings.enumValue( QStringLiteral( "/qgis/defaultProjectFileFormat" ), QgsProject::FileFormat::Qgz );
+    if ( defaultProjectFileFormat == QgsProject::FileFormat::Qgs )
+    {
+      exts = qgsExt + ";;" + zipExt;
```
Comment (m-kuhn) [inline src/app/qgisapp.cpp]:
```suggestion
      exts = qgsExt + QStringLiteral( ";;" ) + zipExt;
```

File: src/app/qgisapp.cpp [hunk 2]
```diff
@@ -6660,12 +6660,23 @@ bool QgisApp::fileSave()
...
+    }
+    else
+    {
+      exts = zipExt + ";;" + qgsExt;
```
Comment (m-kuhn) [inline src/app/qgisapp.cpp]:
```suggestion
      exts = zipExt + QStringLiteral( ";;" ) + qgsExt;
```

**Review by m-kuhn — APPROVED** [review]
Good move, I see a lot of people always switching this manually every time

**Comment by elpaso** [comment]
@m-kuhn I suspect they are mostly developers, but it's a good move anyway. I wonder if the sidecar db could be loaded anyway following a same-basename convention.

**Comment by 3nids** [comment]
> I wonder if the sidecar db could be loaded anyway following a same-basename convention

that's a good idea!

**Comment by m-kuhn** [comment]
> I suspect they are mostly developers

They are the GIS experts inside companies / organizations.
With developers I noticed they are too lazy to do extra clicks unless there is an urgent need.

> I wonder if the sidecar db could be loaded anyway following a same-basename convention.

:+1:

**Pull request merged**